src/AutoRest.CSharp/Common/Generation/Writers/CodeWriter.cs

```diff
@@ -36,12 +36,14 @@ public CodeWriter()
             _scopes.Push(new CodeWriterScope(this, "", false));
         }
 
-        public CodeWriterScope Scope(FormattableString line, string start = "{", string end = "}", bool newLine = true)
+        public CodeWriterScope Scope(FormattableString line, string start = "{", string end = "}", bool newLine = true, CodeWriterScopeDeclarations? scopeDeclarations = null)
```

Comment:
@archerzz This is a part you can grab into the https://github.com/Azure/autorest.csharp/pull/2554 (or just wait until this one is merged).

Reply:
Thanks. I'd like to wait for this to merge in, #2554 is not a high priority.